### PR TITLE
Implement new rpc related functionality

### PR DESCRIPTION
**Pull request by chippmann**
This changes our rpc related code to support the changes made in godot.

Namely;
- Drop of rset and any other property related rpc calls and configuration
- New rpc config and config retrieval

This also updates the registration code, code generation and the IDE checks.

### CODE REVIEW

File: harness/tests/src/main/kotlin/godot/tests/FuncRefTest.kt [hunk 1]
```diff
@@ -16,7 +16,7 @@ class FuncRefTest : Node() {
     @RegisterSignal
     val signalTest by signal()
 
-    @RegisterProperty(rpcMode = MultiplayerAPI.RPCMode.REMOTE)
+    @RegisterProperty
     var blubb: Boolean = false
```
Comment (piiertho) [inline harness/tests/src/main/kotlin/godot/tests/FuncRefTest.kt:20]: Isn't it missing `@Rpc` annotation ?

File: harness/tests/src/main/kotlin/godot/tests/Invocation.kt [hunk 2]
```diff
@@ -332,7 +332,7 @@ class Invocation : Spatial() {
 	override fun _ready() {
 		val formerName = name
 		println("Name is: $name")
-		name = "TestName"
+		name = StringName("TestName") //TODO/4.0: provide convenience wrapper for StringName properties and parameters
```
Comment (piiertho) [inline harness/tests/src/main/kotlin/godot/tests/Invocation.kt:335]: You should have an extension method to convert `String` to `StringName`

File: harness/tests/src/main/kotlin/godot/tests/rpctests/RpcTests.kt [hunk 3]
```diff
@@ -1,33 +1,25 @@
...
 
 @RegisterClass("RPCTests")
 class RpcTests : Node() {
 
     @RegisterProperty
     var remoteSyncCalled: Boolean = false
 
-    @RegisterProperty(rpcMode = MultiplayerAPI.RPCMode.REMOTESYNC)
+    @RegisterProperty
     var remoteSyncProperty: Boolean = false
```
Comment (piiertho) [inline harness/tests/src/main/kotlin/godot/tests/rpctests/RpcTests.kt:13]: Missing `@Rpc` annotation ?

File: kt/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/ext/TypeExtensions.kt [hunk 4]
```diff
@@ -25,7 +25,6 @@ fun Type?.toKtVariantType(): ClassName = when {
         "godot.core.VariantType",
         fqName.substringAfterLast(".").camelToSnakeCase().uppercase(Locale.getDefault())
     )
-    fqName == "kotlin.Unit" -> ClassName("godot.core.VariantType", "NIL")
```
Comment (piiertho) [inline kt/entry-generation/godot-entry-generator/src/main/kotlin/godot/entrygenerator/ext/TypeExtensions.kt:28]: Not sure why this is removed

File: src/kt_class.cpp [hunk 5]
```diff
@@ -99,6 +99,15 @@ void KtClass::fetch_methods(jni::Env& env) {
     }
 }
 
+void KtClass::fetch_rpc_methods() {
+    HashMap<StringName, KtFunction*>::Iterator current = methods.begin();
+    while (current) {
```
Comment (piiertho) [inline src/kt_class.cpp:104]: With 4.0 you can now use classic for each, like here : https://github.com/utopia-rise/godot-kotlin-jvm/blob/d108e232d4b0599ba53aca3d54ed382e653fdcb2/src/gd_kotlin.cpp#L456

File: src/kotlin_script.cpp [hunk 6]
```diff
@@ -205,8 +205,7 @@ void KotlinScript::set_path(const String& p_path, bool p_take_over) {
 }
 
 const Vector<Multiplayer::RPCConfig> KotlinScript::get_rpc_methods() const {
```
Comment (piiertho) [inline src/kotlin_script.cpp:207]: Seems method changed recently: https://github.com/godotengine/godot/commit/ca7d572908c58c587214b8f65bdd4078d0185ab2

**Review by piiertho — CHANGES_REQUESTED** [review]
Great work !
I have some questions, and new commit on godot master invalidate cpp code.

**Comment by chippmann** [comment]
Superseeded by #369

**Pull request closed without merging**